roles/matrix-ldap-registration-proxy/defaults/main.yml

```diff
@@ -0,0 +1,47 @@
+# Project source code URL: https://gitlab.com/activism.international/matrix_ldap_registration_proxy
+
+matrix_ldap_registration_proxy_enabled: false
+
+matrix_ldap_registration_proxy_container_image_self_build_repo: "https://gitlab.com/activism.international/matrix_ldap_registration_proxy.git"
+matrix_ldap_registration_proxy_container_image_self_build_branch: "{{ matrix_ldap_registration_proxy_version }}"
+
+matrix_ldap_registration_proxy_version: "296246afc6a9b3105e67fcf6621cf05ebc74b873"
+
+matrix_ldap_registration_proxy_base_path: "{{ matrix_base_data_path }}/matrix_ldap_registration_proxy"
+# We need the docker src directory to be named matrix_ldap_registration_proxy.
+matrix_ldap_registration_proxy_docker_src_files_path: "{{ matrix_ldap_registration_proxy_base_path }}/docker-src/matrix_ldap_registration_proxy"
+matrix_ldap_registration_proxy_config_path: "{{ matrix_ldap_registration_proxy_base_path }}/config"
+
+matrix_ldap_registration_proxy_ldap_uri: "{{ matrix_synapse_ext_password_provider_ldap_uri }}"
+matrix_ldap_registration_proxy_ldap_base_dn: "{{ matrix_synapse_ext_password_provider_ldap_base }}"
+matrix_ldap_registration_proxy_ldap_user: "{{ matrix_synapse_ext_password_provider_ldap_bind_dn }}"
+matrix_ldap_registration_proxy_ldap_password: "{{ matrix_synapse_ext_password_provider_ldap_bind_password }}"
+matrix_ldap_registration_proxy_matrix_server_name: "{{ matrix_domain }}"
+matrix_ldap_registration_proxy_matrix_server_url: "https://{{ matrix_server_fqn_matrix }}"
+
+# Controls whether the self-check feature should validate SSL certificates.
+matrix_matrix_ldap_registration_proxy_self_check_validate_certificates: true
+
+matrix_ldap_registration_proxy_container_port: 8080
+# Controls whether the matrix_ldap_registration_proxy container exposes its HTTP port (tcp/{{ matrix_ldap_registration_proxy_container_port }} in the container).
+#
+# Takes an "<ip>:<port>" or "<port>" value (e.g. "127.0.0.1:8080"), or empty string to not expose.
+matrix_ldap_registration_proxy_container_http_host_bind_port: ''
+
+# A list of extra arguments to pass to the container
+matrix_ldap_registration_proxy_container_extra_arguments: []
+
+# List of systemd services that matrix_ldap_registration_proxy.service depends on
+matrix_ldap_registration_proxy_systemd_required_services_list: ['docker.service']
+
+# List of systemd services that matrix_ldap_registration_proxy.service wants
+matrix_ldap_registration_proxy_systemd_wanted_services_list: []
+
+# Default ma1sd configuration template which covers the generic use case.
+# You can customize it by controlling the various variables inside it.
+matrix_ldap_registration_proxy_configuration_env: "{{ lookup('template', 'templates/ldap-registration-proxy.env.j2') }}"
+
+# Holds the final ma1sd configuration (a combination of the default and its extension).
+matrix_ldap_registration_proxy_configuration: "{{ matrix_ldap_registration_proxy_configuration_env }}"
```

Comment:
There are some `ma1sd` leftovers here.

Also, it may be unnecessary to go through such a complicated process (`template`, etc.) for some `env` file.

In other roles where we have env files like this, we usually offer some `_extension` variable which lets people append additional environment variables to the file. All existing environment variables (part of the default `env` file) can be redefined using playbook variables (like `matrix_ldap_registration_proxy_ldap_base_dn`, for example).

You can take inspiration from the `matrix-jitsi` role.

https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/5fafbab7ae4b6fd090c152144932da6e57f883cd/roles/matrix-jitsi/defaults/main.yml#L162-L172

https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/5fafbab7ae4b6fd090c152144932da6e57f883cd/roles/matrix-jitsi/templates/web/env.j2#L110